### PR TITLE
Make CommonDetailGrid mobile-friendly

### PR DESCRIPTION
**Pull request by martha**
## Description

PT ticket: https://www.pivotaltracker.com/story/show/187296491
This PR makes the changes described in the figma [here](https://www.figma.com/file/F9EHR2IZYuMAlOJ9aOG2IL/OP-%2F-HMIS-%2F-Phase-2-(2024)?type=design&node-id=2223-24078&mode=design&t=DawrICJx3X8lRXqc-0) for making the Details card more mobile-friendly.

Before:
Screenshot 2024-03-26 at 12 43 14 PM
After:
Screenshot 2024-03-26 at 12 39 21 PM

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/components/elements/CommonDetailGrid.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 import React, { ReactNode, useId } from 'react';
+import { useIsMobile } from '@/hooks/useIsMobile';
 
 // Grid of Key/Value pairs that has the specific styling used for Enrollment Details and custom client attributes
 
 export const CommonDetailGridContainer: React.FC<{
   children: ReactNode;
 }> = ({ children }) => {
+  const isTiny = useIsMobile('sm');
```
Comment (gigxz) [inline src/components/elements/CommonDetailGrid.tsx]: I think we want to use`md` here to match the behavior of the `CommonDetailGridItem`s. It looks like there is a point between large and small where the table wraps, but the borders remain:

Screenshot 2024-03-26 at 2 50 31 PM

Comment (gigxz) [inline src/components/elements/CommonDetailGrid.tsx]:
```suggestion
  const isTiny = useIsMobile('md');
```

**Review by gigxz — APPROVED** [review]
👍

File: src/components/elements/CommonDetailGrid.tsx (hunk 1 above)
Comment (martha) [reply src/components/elements/CommonDetailGrid.tsx]: Thank you!

**Pull request merged**